pgml-sdks/pgml/src/builtins.rs

```diff
@@ -87,6 +88,55 @@ impl Builtins {
         let results = results.first().unwrap().get::<serde_json::Value, _>(0);
         Ok(Json(results))
     }
+
+    /// Run the built-in `pgml.embed()` function.
+    ///
+    /// # Arguments
+    ///
+    /// * `model` - The model to use.
+    /// * `text` - The text to embed.
+    ///
+    pub async fn embed(&self, model: &str, text: &str) -> anyhow::Result<Json> {
+        let pool = get_or_initialize_pool(&self.database_url).await?;
+        let query = sqlx::query("SELECT embed FROM pgml.embed($1, $2)");
+        let result = query.bind(model).bind(text).fetch_one(&pool).await?;
+        let result = result.get::<Vec<f32>, _>(0);
+        let result = serde_json::to_value(result)?;
+        Ok(Json(result))
+    }
+
+    /// Run the built-in `pgml.embed()` function, but with handling for batch inputs and outputs.
+    ///
+    /// # Arguments
+    ///
+    /// * `model` - The model to use.
+    /// * `texts` - The texts to embed.
+    ///
+    pub async fn embed_batch(&self, model: &str, texts: Json) -> anyhow::Result<Json> {
+        let texts = texts
+            .0
+            .as_array()
+            .with_context(|| "embed_batch takes an array of texts")?
+            .into_iter()
+            .map(|v| {
+                v.as_str()
+                    .with_context(|| "only text embeddings are supported")
+                    .unwrap()
```

Comment:
You can collect into a `Result<Vec<_>>` so you can `?` the `Result` instead of panicking if the value is not a `str`.

Reply:
Something like

```rust
let texts = texts
            .0
            .as_array()
            .with_context(|| "embed_batch takes an array of texts")?
            .iter()
            .map(|v| {
                v.as_str()
                    .with_context(|| "only text embeddings are supported")
                    .map(|s| s.to_string())
            })
            .collect::<anyhow::Result<Vec<String>>>()?;
```

I think will work.

Reply:
I don't think we want that. The API accepts an array of strings, so if someone passes in an array of objects that can be cast to a string, they will get embeddings of integers, or whatever else they passed in, which will make them think that everything is fine.

Reply:
We are still calling `as_str` here so if it was an object, it wouldn't be treated as a string. I was merely suggesting using `Result::Err` instead of `unwrap`. But all is good.